{{cookiecutter.project_slug}}/exp1/steps.py

```diff
@@ -0,0 +1,59 @@
+#%%
+from matplotlib import pyplot as plt
+import numpy as np
+from art.step.steps import ExploreData
+from art.step.step_savers import MatplotLibSaver
+from collections import Counter
+
+
+class DataAnalysis(ExploreData):
+    """This step allows you to perform data analysis and extract information that is necessery in next steps"""
+
+    def do(
+            self, previous_states
+    ):
+        targets = []
+
+        # Loop through batches in the mnist_data_module train dataloader
+        for batch in self.datamodule.train_dataloader():
```

Comment:
Okay, so we stick with providing the user with everything he needs. I understand, that in the future all `X = Y` will be replaced with `X = ...`?